src/components/shared/TinderCard/index.tsx

```diff
@@ -0,0 +1,211 @@
+  onSwipeLeft?: (item: DataProps) => void;
+  onCancel?: () => void;
+  data: Array<DataProps>;
+  renderCards: (item: DataProps) => ReactElement;
+  renderNoMoreCards: () => ReactElement;
+  rotate?: boolean;
+  stackSize?: number;
+}
+
+const Container = styled.View`
+  width: ${SCREEN_WIDTH};
+`;
+
+const CancelButton = styled.Button`
+  background-color: blue;
+  color: blue;
+`;
+
+const NoCard = styled.View`
+  width: 100%;
+  height: 200px;
+  justify-content: center;
+  align-items: center;
+  padding: 10px;
+`;
+
+function TinderCard(props: Props): ReactElement {
+  const [cardIndex, setCardIndex] = useState(0);
+  const position = useMemo(() => new Animated.ValueXY(), []);
+
+  const resetPosition = (): void => {
+    Animated.spring(position, {
+      toValue: { x: 0, y: 0 },
+    }).start();
+  };
+
+  const onSwipeCompleted = useCallback((direction: string): void => {
+    const { onSwipeLeft, onSwipeRight, data } = props;
+
+    position.setValue({ x: 0, y: 0 });
+
+    UIManager.setLayoutAnimationEnabledExperimental &&
+      UIManager.setLayoutAnimationEnabledExperimental(true);
+    LayoutAnimation.spring();
+
+    let currentIndex = 0;
+
+    setCardIndex((idx) => {
+      currentIndex = idx;
+      return idx + 1;
+    });
+
+    if (direction === 'right') {
+      onSwipeRight && onSwipeRight(data[currentIndex]);
+    } else {
+      onSwipeLeft && onSwipeLeft(data[currentIndex]);
+    }
+  }, []);
+
+  const forceSwipe = (direction: string): void => {
+    const x = direction === 'right' ? SCREEN_WIDTH : -SCREEN_WIDTH;
+    Animated.timing(position, {
+      toValue: { x, y: 0 },
+      duration: SWIPE_OUT_DURATION,
+    }).start(() => onSwipeCompleted(direction));
+  };
+
+  const _panResponder = useMemo(
+    () =>
+      PanResponder.create({
+        // Ask to be the responder:
+        onStartShouldSetPanResponder: (evt, gestureState) => true,
+        onPanResponderMove: (evt, gestureState) => {
+          position.setValue({ x: gestureState.dx, y: gestureState.dy });
+        },
+        onPanResponderRelease: (evt, gesture) => {
+          if (gesture.dx > SWIPE_THRESHOLD) {
+            forceSwipe('right');
+          } else if (gesture.dx < -SWIPE_THRESHOLD) {
+            forceSwipe('left');
+          } else {
+            resetPosition();
+          }
+        },
+      }),
+    [],
+  );
+
+  const getCardStyle = (): {} => {
+    const rotate = position.x.interpolate({
+      inputRange: [-SCREEN_WIDTH * 1.5, 0, SCREEN_WIDTH * 1.5],
+      outputRange: ['-45deg', '0deg', '45deg'],
+    });
+
+    const rotateStyle = props.rotate ? { transform: [{ rotate }] } : {};
+
+    return {
+      ...position.getLayout(),
+      ...rotateStyle,
+    };
+  };
+
+  const _renderCards = (): ReactElement | ReactElement[] => {
+    if (!props.data || cardIndex >= props.data.length) {
+      return props.renderNoMoreCards();
+    }
+
+    const dataSet = props.data.map((item, idx) => {
+      if (idx < cardIndex) return null;
```

Comment:
Do you think we'll face such a case like this? This may lead to a dangerous side effects.